doc/user/content/sql/create-source/postgres.md

```diff
@@ -0,0 +1,166 @@
+
+If you stop or delete Materialize without first dropping the Postgres source, the Postgres replication slot isn't deleted and will continue to accumulate data. In such cases, you should manually delete the Materialize replication slot to recover memory and avoid degraded performance in the upstream database. Materialize replication slot names always begin with `materialize_` for easy identification.
+
+#### Restrictions on Postgres sources
+
+- Materialize does not support changes to schemas for existing publications. You will need to drop the existing sources and then recreate them after creating new publications for the updated schemas.
+- Sources can only be created from publications that use [data types](/sql/types/) supported by Materialize. Attempts to create sources from publications which contain unsupported data types will fail with an error.
+- Materialize only supports [TOASTED](https://www.postgresql.org/docs/current/storage-toast.html) values for append-only tables. Practically speaking, you can include rows with TOASTED values as long as they are never updated or deleted, or you can disable TOAST on the original Postgres table. If a TOASTED column is updated, the source will enter an errored state that renders all per-table views inaccessible.
+
+  To disable TOAST on a column, use the following command in Postgres:
+  ```
+  ALTER TABLE table_name ALTER COLUMN column_name
+  SET STORAGE PLAIN;
+  ```
+- Tables replicated into Materialize should not be truncated. If a table is truncated while replicated, the whole source becomes inaccessible and will not produce any data until it is re-created.
+
+## Examples
+
+### Setting up PostgreSQL
+
+Before you create a Postgres source in Materialize, you must complete the following prerequisite steps in Postgres.
+
+1. Ensure the database allows logical replicationz ( in the configuration). For most configurations, it should suffice to set `wal_level = logical` in `postgresql.conf`, but additional steps may be required for Amazon RDS. See the [Amazon Relational Database Service Documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.FeatureSupport.LogicalReplication) for details.
+
+2. Assign the user `REPLICATION` privileges:
+    ```sql
+    ALTER ROLE "user" WITH REPLICATION;
+    ```
+3. Set replica identity to full:
+    ```sql
+    ALTER TABLE foo
+    REPLICA IDENTITY FULL;
+    ```
+4. Create a publication containing all the tables you wish to query in Materialize:
+
+    *For all tables in Postgres:*
+    ```sql
+    CREATE PUBLICATION mz_source FOR ALL TABLES;
+    ```
+
+    *For specific tables:*
+    ```sql
+    CREATE PUBLICATION mz_source FOR TABLE table1, table2;
+    ```
+
+### Creating a source
+
+Once you have set up Postgres, you can create the source in Materialize.
+
+```sql
+CREATE MATERIALIZED SOURCE "mz_source" FROM POSTGRES
+HOST 'host=postgres port=5432 user=host sslmode=disable dbname=postgres'
```

Comment:
Since many users will be copy-pasting these commands let's remove the `sslmode=disable` to keep the default setting which will attempt to encrypt the connection if possible